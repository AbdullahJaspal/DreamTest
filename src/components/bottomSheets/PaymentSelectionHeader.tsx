import {
  Dimensions,
  Image,
  Pressable,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import React, {useState} from 'react';
import {icons} from '../../assets/icons';
const {width} = Dimensions.get('screen');

const PaymentSelectionHeader: React.FC = () => {
  const [showInfo, setShowInfo] = useState(false);

  return (
    <View style={styles.upper_container}>
      <Text style={styles.recharge_text}>Payments Method</Text>

      <Pressable
        style={styles.question_mark_view}
        onPress={() => setShowInfo(true)}>
        <Image source={icons.questionMark} style={{width: 20, height: 20}} />
      </Pressable>
      {/* <InfoModal
        show_model={showInfo}
        setShowModel={setShowInfo}
        title="Payment Method Info"
      /> */}
    </View>
  );
};

export default PaymentSelectionHeader;

const styles = StyleSheet.create({
  upper_container: {
    width: width,
    alignItems: 'center',
    justifyContent: 'center',
    borderBottomWidth: 0.5,
    padding: 15,
    flexDirection: 'row',
  },
  recharge_text: {
    color: 'black',
    fontWeight: '600',
    fontSize: 16,
  },
  question_mark_view: {
    position: 'absolute',
    right: 15,
  },
});
