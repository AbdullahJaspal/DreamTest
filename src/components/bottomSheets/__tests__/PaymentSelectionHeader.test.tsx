import 'react-native';
import React from 'react';

import {it} from '@jest/globals';
import renderer from 'react-test-renderer';
import PaymentSelectionHeader from '../PaymentSelectionHeader';

it('Icon renders correctly', () => {
  const tree = renderer.create(<PaymentSelectionHeader />).toJSON();

  expect(tree).toMatchSnapshot();
});
