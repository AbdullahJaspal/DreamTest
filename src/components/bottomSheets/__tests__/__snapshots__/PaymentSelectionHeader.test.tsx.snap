// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 15,
      "width": 750,
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    Payments Method
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "position": "absolute",
        "right": 15,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/images/question_mark.png",
        }
      }
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
